### DISCUSSION TITLE
modules/network/mqtt vs. embedded:network/mqtt

### DISCUSSION THREAD

**Discussion by cmidgley** [Moddable-OpenSource/moddable#1366] [General]
I've been using the original `modules/network/mqtt` client (modified heavily to package size issues) but I'm considering using the newer `io` version.  I see the TypeScript typings are available in `typings/embedded_network/mqtt`, implying that it is a native service of Moddable. However, the implementation and `manifest_mqttclient.json` are under `examples/io/tcp/mqttclient`, and `examples` seems an unusual location for a supported service (also `manifest.json` includes a `main.js`, further demonstrating it is an example).

What is the recommended MQTT client to use?  Is the `examples` version considered supported and stable, or are there issues still to be resolved?

**Comment by phoddie** [comment]
Long delayed reply.... `embedded:network/mqtt` is preferred. The implementation will eventually migrate. There aren't known issues. If issues are identified, we're much more motivated to resolve them in the 419 implementation than the original mqtt module.
